agents/demo/demo.ipynb

```diff
@@ -0,0 +1,854 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Configuring Kubernetes credentials"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Having done so we are ready to obtain the credentials for the Kubernetes cluster on which we will be running the training tasks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 119,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PROJECT=\"kubeflow-rl\"\n",
+    "ZONE=\"us-east1-d\"\n",
+    "CLUSTER=\"name-of-your-cluster\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!gcloud container clusters --project={PROJECT} --zone={ZONE} get-credentials {CLUSTER}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Configure ksonnet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Almost there! The last thing we need to do is configure our ksonnet environment to use these kube credentials by default:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.chdir('/home/jovyan/kubeflow-examples/agents')\n",
+    "\n",
+    "!ks env add default"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The objective of the training phase is to learn the parameterization of our model that confers a high level of performance on the provided task. Here we'll launch and monitor a job."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Launching the TFJob"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll use [ksonnet](https://ksonnet.io/) to parameterize and apply a TFJob configuration (i.e. run a job). Here you can change the image to be a custom job image, such as one built and deployed with build.sh, or use the one provided here if you only want to change parameters. Below we'll display the templated job YAML for reference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check your cluster and see if that matches one of the existing ksonnet environments\n",
+    "# You want the kubernetes master server to be the same as the server listed for the ks environment\n",
+    "!kubectl cluster-info\n",
+    "!ks env list"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 108,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "gs://kubeflow-rl-kf/studies/replicated-env-comparison\n"
+     ]
+    }
+   ],
+   "source": [
+    "image = \"gcr.io/kubeflow-rl/agents:tf1.4.1-0208-0959-a276\"\n",
```

Comment:
This image is not publicly available. Guessing I should run `demo/build.sh`?

Proposed changes:

1. Add instructions to build image here. (Maybe let users call build script from notebook? Then again, this should be part of the setup, not the demo.)

2. Remove the hardcoding in the notebook.

Reply:
Indeed an issue and re your other comment there should be both a public image as well as instructions to build and use a locally built image.